### ISSUE TITLE
Add no-useless-interpolation

### ISSUE THREAD

**Issue by vsemozhetbyt** [eslint/eslint#10798]
**Please describe what the rule should do:**

This rule is like [`no-useless-concat`](https://eslint.org/docs/rules/no-useless-concat) but for template literals. It prevents literal into literal interpolation that seems redundant.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[X] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const str1 = `abc${'def'}ghi`;
const str2 = `123${456}759`;
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It seems this is in line with `no-useless-concat` rule.

**Comment by ilyavolodin** [comment]
At the first glance, it seems that the scope of the proposal is very limited to be included as a core rule, although it's still a valuable check, just focused on a very small subset of edge cases.

**Comment by kaicataldo** [comment]
It's a bit hard to evaluate this without some more details about what actually triggers an error. Some questions I have:
 - Would this only be string or number literals? 
 - What happens in the case of nested cases, such as 
```js
`123${`abc${'def'}ghi`}759`
```
 - And what about cases like the following?
```js
const foo = '123';
const bar = `123${foo}456`;
```

**Comment by bendtherules** [comment]
In my opinion (hope OP doesn't mind) :
1. Only strings. Doing `.toString()` of other things statically sounds bit sketchy to me.
2. Nested cases are taken care of (depth-first simplification)
3. Don't track variables - values might change over time, can be used in other places, lot of other possibilities.
